### ISSUE TITLE
elvish -c does not accept "edit:*" commands

### ISSUE THREAD

**Issue by ajeetdsouza** [elves/elvish#1299]
I want to use Elvish in my CI to check for syntax errors and the like. However, I'm unable to do this, because passing the source code to `elvish -c` throws the following error:

```text
compilation error: cannot find variable $edit:completion:arg-completer
```

Is there another way to do this? I tried writing to a file and running `elvish file.elv` but I got the same error.

**Comment by xiaq** [comment]
The editor is currently only initialized in interactive mode. I repurposed this issue to track the ability to start the editor from a script.

**Comment by ajeetdsouza** [comment]
Fair enough. I'm not sure if it's unreasonable to expect the editor to start up inside a script, but I'd expect at least the `-compileonly` flag to work in this case.

**Comment by krader1961** [comment]
In the recent past I have seen a handful of questions on IRC, and Github issues, asking that the `edit` module be accessible in a non interactive shell context. None of which have justified exposing the `edit` namespace in a non-interactive context.

@ajeetdsouza: Can you provide more context on why you are using commands in a non-interactive context that only make sense in an interactive context? How should the various `edit` module commands behave in a non-interactive context?

**Comment by krader1961** [comment]
> but I'd expect at least the -compileonly flag to work in this case.

Why do you expect that to work? You asked Elvish to compile code that is only valid in an interactive context in a non interactive context. This is the sort of thing where the `script` command is typically used.

**Comment by ajeetdsouza** [comment]
My main use case is - I have Elvish completion scripts in my repository, and I'd like to add a workflow to my CI to check for invalid syntax. Since it's a completion script, it has `edit:` commands.

**Comment by krader1961** [comment]
> My main use case is - I have Elvish completion scripts in my repository, and I'd like to add a workflow to my CI to check for invalid syntax.

Not surprisingly, your use case is the one that sticks in my mind from a nearly identical discussion thread on IRC a few months ago. That discussion should have resulted in an equivalent issue being opened so that this one could be closed as a duplicate. 😄 

The challenge is finding an elegant way to make the interactive `edit` module useful in non-interactive Elvish scripts without creating other problems; e.g., panics from `edit` module code because the shell is not actually interactive.

**Comment by ajeetdsouza** [comment]
> The challenge is finding an elegant way to make the interactive edit module useful in non-interactive Elvish scripts without creating other problems; e.g., panics from edit module code because the shell is not actually interactive.

@krader1961 shouldn't most `edit` commands be a no-op in a non-interactive setting? Or is there something that might break as a result?
